autoload/lsp/internal/work_done_progress.vim

```diff
@@ -0,0 +1,68 @@
+" https://microsoft.github.io/language-server-protocol/specifications/specification-current/#progress
+
+let s:progress_ui = []
+let s:enabled = 0
+
+function! lsp#internal#work_done_progress#_enable() abort
+    if !g:lsp_work_done_progress_enabled | return | endif
+
+    if s:enabled | return | endif
+    let s:enabled = 1
+
+    let s:Dispose = lsp#callbag#pipe(
+          \ lsp#stream(),
+          \ lsp#callbag#filter({x->has_key(x, 'response') && has_key(x['response'], 'method')
+          \  && x['response']['method'] ==# '$/progress' && has_key(x['response'], 'params')
+          \  && has_key(x['response']['params'], 'value') && type(x['response']['params']['value']) == type({})}),
+          \  lsp#callbag#subscribe({'next': {x->s:handle_work_done_progress(x['server'], x['response'])}})
+          \ )
+endfunction
+
+function! s:handle_work_done_progress(server, response) abort
+    let l:value = a:response['params']['value']
+    " Add the server name to distinguish the server
+    let l:token = a:server . ':' . a:response['params']['token']
```

Comment:
can we avoid this and instead in filter check both token and server?

Reply:
No.
When the same token is received from different servers, both should be handled correctly.
Therefore, we cannot filter it.

Reply:
I mean why not do something like this so you the token is the same as the original one.

```vim
let s:progress_ui = filter(s:progress_ui, {_, x->x['token'] !=# l:token && x['server'] !=# a:server })
```

Let me know if I'm missing something.

Reply:
sorry. I was confused with callbag#filter.
I fixed it.